src/OracleService/OracleService.cs

```diff
@@ -345,9 +345,13 @@ private async Task<(OracleResponseCode, string)> ProcessUrlAsync(string url)
                 return (OracleResponseCode.Error, null);
             if (!protocols.TryGetValue(uri.Scheme, out IOracleProtocol protocol))
                 return (OracleResponseCode.ProtocolNotSupported, null);
+
+            using CancellationTokenSource ctsTimeout = new(Settings.Default.MaxOracleTimeout);
+            using CancellationTokenSource ctsLinked = CancellationTokenSource.CreateLinkedTokenSource(cancelSource.Token, ctsTimeout.Token);
```

Comment:
Can these two tokens merge to single one?

Reply:
We want to cancel it if `cancelSource` was cancel, so we need to join them